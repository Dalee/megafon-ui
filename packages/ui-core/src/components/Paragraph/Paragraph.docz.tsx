import * as React from 'react';

export const DemoMarginWrapper: React.FC = ({ children }) => (
    <div style={{ margin: 16 }}>{children}</div>
);

export const DemoBackgroundWrapper: React.FC = ({ children }) => (
    <div style={{ background: '#00B956' }}>
        {children}
    </div>
);
