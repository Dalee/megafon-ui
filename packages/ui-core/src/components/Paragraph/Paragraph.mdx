---
name: Paragraph
menu: Components
route: components/paragraph
---

import { Playground, Props } from 'docz';
import { DemoMarginWrapper, DemoBackgroundWrapper } from './Paragraph.docz';
import Paragraph from './Paragraph';

# Paragraph

## Properties

<Props of={Paragraph} />

## Basic usage

<Playground>
    <DemoMarginWrapper>
        <Paragraph>
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Odit dolore
            quam deserunt magni consequuntur veniam! Enim doloremque error expedita
            nisi nobis architecto aliquid. Iste dolores cupiditate earum vero, at
            quis.
        </Paragraph>
    </DemoMarginWrapper>
</Playground>

## Usage without margin

<Playground>
    <DemoMarginWrapper>
        <Paragraph hasMargin={false}>
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Odit dolore
            quam deserunt magni consequuntur veniam! Enim doloremque error expedita
            nisi nobis architecto aliquid. Iste dolores cupiditate earum vero, at
            quis.
        </Paragraph>
        <Paragraph hasMargin={false}>
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Odit dolore
            quam deserunt magni consequuntur veniam! Enim doloremque error expedita
            nisi nobis architecto aliquid. Iste dolores cupiditate earum vero, at
            quis.
        </Paragraph>
    </DemoMarginWrapper>
</Playground>

## Align options

<Playground>
    <DemoMarginWrapper>
        <Paragraph align='left'>
            left - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
        <Paragraph align='center'>
            center - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
        <Paragraph align='right'>
            right - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
    </DemoMarginWrapper>
</Playground>

## Size options

<Playground>
    <DemoMarginWrapper>
        <Paragraph>
            15/24 - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
        <Paragraph size={'small'}>
            12/16 - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
    </DemoMarginWrapper>
</Playground>

## Color options

<Playground>
    <DemoMarginWrapper>
        <DemoBackgroundWrapper>
            <Paragraph color='clearWhite'>
                'clearWhite' - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
            </Paragraph>
            <Paragraph color='spbSky0'>
                'spbSky0' - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
            </Paragraph>
            <Paragraph color='spbSky1'>
                'spbSky1' - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
            </Paragraph>
            <Paragraph color='spbSky2'>
                'spbSky2' - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
            </Paragraph>
        </DemoBackgroundWrapper>
        <Paragraph color='green'>
            'green' - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
        <Paragraph color='purple'>
            'purple' - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
        <Paragraph color='freshAsphalt'>
            'freshAsphalt' - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
        <Paragraph color='fullBlack'>
            'fullBlack' - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
        <Paragraph color='inherit'>
            'inherit' - Lorem ipsum, dolor sit amet consectetur adipisicing elit.
        </Paragraph>
    </DemoMarginWrapper>
</Playground>
