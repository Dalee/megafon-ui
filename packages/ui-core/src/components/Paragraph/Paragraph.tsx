import * as React from 'react';
import * as PropTypes from 'prop-types';
import './Paragraph.less';
import cnCreate from 'utils/cnCreate';

export interface IParagraphProps {
    /** Align text horizontally */
    align?: 'left' | 'center' | 'right';
    /** Font size */
    size?: 'regular' | 'small';
    /** Vertical margin (enabled by default) */
    hasMargin?: boolean;
    /** Text color */
    color?: 'green' | 'purple' | 'clearWhite' | 'spbSky0' | 'spbSky1' | 'spbSky2' | 'freshAsphalt' | 'fullBlack';
    /** Custom className */
    className?: string;
    children?: React.ReactNode;
}

const cn = cnCreate('mfui-paragraph');
const Paragraph: React.FC<IParagraphProps> = ({
        size,
        align,
        color,
        children,
        className,
        hasMargin,
    }: IParagraphProps) => (
    <p
        className={cn({
            size,
            align,
            color,
            'has-margin': hasMargin,
        }, className)}>
        {children}
    </p>
);

Paragraph.propTypes = {
    align: PropTypes.oneOf(['left', 'center', 'right']),
    size: PropTypes.oneOf(['regular', 'small']),
    hasMargin: PropTypes.bool,
    color: PropTypes.oneOf(['green', 'purple', 'clearWhite', 'spbSky0', 'spbSky1', 'spbSky2', 'freshAsphalt', 'fullBlack']),
    children: PropTypes.node,
};

Paragraph.defaultProps = {
    hasMargin: true,
    size: 'regular',
    color: 'freshAsphalt',
};

export default Paragraph;
