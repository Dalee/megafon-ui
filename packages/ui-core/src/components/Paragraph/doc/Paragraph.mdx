---
name: Paragraph
menu: Компоненты
route: components/paragraph
---

import { Playground, Props } from 'docz';
import Paragraph from '../Paragraph';
import Example from './Paragraph.example.mdx';

# Paragraph

<ImportExample name="Paragraph" />

<DoczTabs>
    <Example />
    <Props of={Paragraph} />
</DoczTabs>
