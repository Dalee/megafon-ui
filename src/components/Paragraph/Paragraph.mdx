---
name: Paragraph
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Paragraph from './Paragraph';

# Paragraph

## Properties

<PropsTable of={Paragraph} />

## Basic usage

<Playground>
    <Paragraph>
        Some text
    </Paragraph>
</Playground>
