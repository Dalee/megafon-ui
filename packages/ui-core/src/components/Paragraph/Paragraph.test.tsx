import * as React from 'react';
import { shallow } from 'enzyme';
import Paragraph, { IParagraphProps } from './Paragraph';

const props: IParagraphProps = {
    align: 'right',
    size: 'small',
    hasMargin: false,
    color: 'purple',
    className: 'test',
    children: 'Test paragraph text',
};

describe('<Paragraph />', () => {
    it('it renders Paragraph with out props', () => {
        const wrapper = shallow(<Paragraph>Text</Paragraph>);
        expect(wrapper).toMatchSnapshot();
    });

    it('it renders Paragraph with props', () => {
        const wrapper = shallow(<Paragraph {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
