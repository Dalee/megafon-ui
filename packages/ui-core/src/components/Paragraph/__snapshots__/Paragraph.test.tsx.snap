// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> it renders Paragraph with out props 1`] = `
<p
  className="mfui-paragraph mfui-paragraph_size_regular mfui-paragraph_color_freshAsphalt mfui-paragraph_has-margin"
>
  Text
</p>
`;

exports[`<Paragraph /> it renders Paragraph with props 1`] = `
<p
  className="mfui-paragraph mfui-paragraph_size_small mfui-paragraph_align_right mfui-paragraph_color_purple test"
>
  Test paragraph text
</p>
`;
